text/0939-import-glob.md

```diff
@@ -0,0 +1,196 @@
+    >
+    > Ultimately this is a concern that goes away once you adopt `.gjs` and that would be our recommendation going forward.
+
+`*` matches everything except:
+ - path separators 
+ - names starting with "." (hidden files)
+
+`**` matches zero or more directories
+
+`?` matches any single character except path separators
+
+`[abc]`: a sequence of characters inside `[]` match any character in that sequence
+
+`{.js,.gjs}`: bash-style brace expansion
+
+### Lazy Loading
+
+`import.meta.glob()` is designed to work nicely in systems that lazily load code. However, it does not promise to *introduce* laziness where laziness does not already exist. 
+
+When building with the classic build pipeline, all your own app code is always included in the bundle regardless of whether anyone imports it or not, and that remains true regardless of whether you use `import.meta.glob()` to access some modules. 
+
+When building with Embroider, you can achieve lazy loading by using `import.meta.glob()` in combination with other features like `staticAppPaths` or `staticComponents`.
+
+The return value from `import.meta.glob()` is the same either way -- you always get functions that return `Promise<Module>`. 
+
+### Replacing cross-package usages
+
+Historically, people have used `requirejs.entries` to have complete global access to everywhere from everywhere. `import.meta.glob` is deliberately more restrictive. For example, an addon cannot use `import.meta.glob` to load code out of the consuming application. Instead, addon authors will need to ask apps to pass them what they need.
+
+For example, a future version of `ember-cli-mirage` might tell app authors to put this code into their app and/or tests as a way to dynamically gain access to all the Mirage-specific models, adapters, serializers, etc that the user has written:
+
+```js
+import { setup } from 'ember-cli-mirage';
+
+setup(import.meta.glob('./mirage/**/*.js'))
+```
+
+Similarly, to do auto-discovery of ember-data models, an existing API like `discoverEmberDataModels()` would now accept them explicitly:
+
+```js
+import { discoverEmberDataModels } from 'ember-cli-mirage';
+
+discoverEmberDataModels(import.meta.glob('../models/*.{js,ts}'));
+```
+
+### Not allowed in publication format
+
+Addons are free to use `import.meta.glob` in their own code, but our tooling should implement it within the addon's own build, *before* publishing to NPM. `import.meta.glob` is not allowed in published addons on NPM.
+
+This greatly reduces future compatibility concerns, and it doesn't cost us anything in terms of flexibility, given that this spec says `import.meta.glob` is not allowed to cross package boundaries anyway. That is: at publish time, the full list of files that any `import.meta.glob()` expands into is statically known.
+
+### Types
+
+`import.meta.glob` has this signature:
+
+```ts
+(pattern: string): Record<string, Promise<unknown>>
```

Comment:
seems Record value should be a function
```
(pattern: string): Record<string, () => Promise<unknown>>
```

Reply:
This is correct, thanks, that's just a bug in my text.